src/components/SignupModal.jsx

```diff
@@ -0,0 +1,180 @@
+import React from "react";
+import { connect } from "react-redux";
+import PropTypes from "prop-types";
+import { TextInput, SubmitButton, MessageBox } from "../views/Form";
+import authUser, { signupError } from "../redux/actions/authUser";
+import { clearError } from "../redux/actions/common";
+
+class SignupModal extends React.Component {
+  state = {
+    user: { username: "", email: "", password: "", confirm_password: "" }
```

Comment:
Refactor user to be in the format
user: {
  username: "",
  email: "",
  ...
}

Reply:
@meshack-mbuvi I understood it as formatting

Reply:
There's no difference really @meshack-mbuvi, the only difference is when we have all that content in one single line it becomes a little difficult to read. My suggestion improves readability and makes the code look a bit cleaner. And no, do not include user in user again.